googletest/test/gtest_env_var_test.py

```diff
@@ -98,6 +100,25 @@ def testEnvVarAffectsFlag(self):
       TestFlag('death_test_use_fork', '1', '0')
       TestFlag('stack_trace_depth', '0', '100')
 
+  def testXmlOutputFile(self):
+    """Test that $XML_OUTPUT_FILE affects the output flag."""
+
+    try:
+      # $XML_OUTPUT_FILE is overridden by $GTEST_OUTPUT
```

Comment:
Can you restructure this test so that each of the scenarios is independent?
Right now each is building on state from the prior scenario.
Maybe refactoring a little with a subfunction would be clearer.